src/test/shared/ui/picker.test.ts

```diff
@@ -640,61 +646,66 @@ describe('IteratingQuickPickController', async () => {
                     },
                 ])
                 resolve()
-            }, 300)
+            }, interval - 15)
         })
+        await clock.nextAsync()
     })
 
     it('appends a no items item', async () => {
         const controller = new picker.IteratingQuickPickController(
             quickPick,
-            new picker.IteratingQuickPickPopulator<string>(
-                () => blankIteratorFn(),
-                val => {
-                    if (val) {
-                        return [{ label: val.toUpperCase() }]
-                    }
-
-                    return []
-                }
-            )
+            new picker.IteratingQuickPickPopulator<string>(() => blankIteratorFn(), converter)
         )
 
         controller.startRequests()
-
-        await new Promise(resolve => {
+        await clock.nextAsync()
+        new Promise(resolve => {
             setTimeout(() => {
                 assert.deepStrictEqual(quickPick.items, [picker.IteratingQuickPickController.NO_ITEMS_ITEM])
                 resolve()
-            }, 300)
+            }, interval - 15)
         })
+        await clock.nextAsync()
     })
 
     it('only appends values from the current refresh cycle', async () => {
         const controller = new picker.IteratingQuickPickController(
             quickPick,
-            new picker.IteratingQuickPickPopulator<string>(
-                () => iteratorFn(),
-                val => {
-                    if (val) {
-                        return [{ label: val.toUpperCase() }]
-                    }
-
-                    return []
-                }
-            )
+            new picker.IteratingQuickPickPopulator<string>(() => iteratorFn(), converter)
         )
 
         controller.startRequests()
+        await clock.nextAsync()
+        await clock.nextAsync()
+        await clock.nextAsync()
+        await clock.nextAsync()
```

Comment:
Rather than "advancing n times", you should be able to use clock.tickAsync(x) which will fire all appropriate timers, making the test more readable (applies earlier in the file too)

Reply:
nextAsync just hits the next thing in the event loop. I'm thinking it's a bit more readable this way, and I was having some issues when advancing exact times where it wasn't adding other promises in the correct places.

This section is also here to specifically run through any executions that may be in the pipeline for the reset statements just to prove that the array at the end is pristine.

I'll reevaluate if this proves to be fluky.